qiskit/aqua/operators/primitive_ops/summed_pauli_op.py

```diff
@@ -0,0 +1,362 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" SummedPauliOp Class """
+
+import logging
+from typing import Dict, List, Optional, Set, Union, cast
+
+import numpy as np
+from scipy.sparse import spmatrix
+
+from qiskit.circuit import Instruction, ParameterExpression
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+from ... import AquaError
+from ..list_ops.summed_op import SummedOp
+from ..list_ops.tensored_op import TensoredOp
+from ..operator_base import OperatorBase
+from .primitive_op import PrimitiveOp
+
+logger = logging.getLogger(__name__)
+
+
+class SummedPauliOp(PrimitiveOp):
```

Comment:
I am just wondering about the name. When I see SummedPauliOp it is quite close to SummedOp which is a ListOp not a primitiveOp. I am wondering whether say PauliSumOp or PauliTableOp, or another name you might come up with that is not quite so close to SummedOp may be better and not perhaps lead to quite so much user confusing if the names are too similar.

Reply:
SummedPauliOp is a subclass of PrimitiveOp, but has some property compatible with ListOp (e.g. oplist).
SummedPauliOp can be used like SummedOp[PauliOp], but as you said this spec is confusing.

Do you think we should have such a compatibility?

Reply:
Certainly having function to say easily go between this and a SummedOp does not seem unreasonable. Hence having compatibility with other aspects of opflow is a good thing. I think by the name my main thought was that users may easily get mistaken into thinking its a ListOp sub-type, which it is not (does not have methods like distribution, combo_fn etc) it does not have 'abelian' on constructor etc, but maybe that will be added later. So not being a ListOp you should not be passing it to any function that said it takes a ListOp as it may fail. Maybe others have some thoughts here.

Reply:
OK, I changed to PauliSumOp.